sleap_nn/data/providers.py

```diff
@@ -82,6 +104,29 @@ def __iter__(self) -> Iterator[Dict[str, torch.Tensor]]:
                 (1, np.abs(self.max_instances - num_instances), nodes, 2), torch.nan
             )
             instances = torch.cat([instances, nans], dim=1)
+            img_height, img_width = image.shape[-2:]
+
+            # pad images to max_height and max_width
+            if self.max_height is not None:  # only if user provides
+                pad_height = self.max_height - img_height
+                pad_width = self.max_width - img_width
+                if pad_height < 0:
+                    raise Exception(
+                        f"Max height {self.max_height} should be greater than the current image height: {img_height}"
+                    )
+                if pad_width < 0:
+                    raise Exception(
+                        f"Max width {self.max_width} should be greater than the current image width: {img_width}"
+                    )
+                image = np.pad(
+                    image,
+                    ((0, 0), (0, 0), (0, pad_height), (0, pad_width)),
+                    mode="constant",
+                ).astype("float32")
```

Comment:
I don't love this being here... Is there a reason we can't do padding as a separate, configurable block? [Like we do in core SLEAP](https://github.com/talmolab/sleap/blob/0134ab443663bf1b1f0287d632dce9fdef80c14b/sleap/nn/data/resizing.py#L257-L469)?